### Load the data

In [1]:
import pandas as pd
df = pd.read_parquet('./rozetka/data_new.parquet')
df

deep_category  \
0                       Маршрутизаторы LB-LINK   
1                      Геймерские кресла Razer   
2                       Геймерские кресла RZTK   
3       Магниты для досок и аксессуары Buromax   
4                     Геймерские кресла Barsky   
...                                        ...   
759864                Корма для грызунов Gimbi   
759865                Корма для грызунов Gimbi   
759866              Корма для грызунов Beaphar   
759867              Корма для грызунов Vitapol   
759868         Корма для грызунов Versele-Laga   

               deep_category_minus_one  \
0                       Маршрутизаторы   
1                    Геймерские кресла   
2                    Геймерские кресла   
3       Магниты для досок и аксессуары   
4                    Геймерские кресла   
...                                ...   
759864              Корма для грызунов   
759865              Корма для грызунов   
759866              Корма для грызунов   
759867              Корма для грызунов   
759868              Корма для грызунов   

                                              description  \
0       Маршрутизатор LB-Link BL-WR450H WiFi Роутер дл...   
1       Идеальная игровая форма Держите соперников в з...   
2       Модель кресла Cyber от украинского бренда RZTK...   
3       Полный комплект Buromax Super Start всех прина...   
4       Удобное кресло из экокожи геймерского дизайна ...   
...                                                   ...   
759864  Продукты GimBi – это высококачественные лакомс...   
759865  Продукты GimBi – это высококачественные лакомс...   
759866  Вкусное и полезное лакомство для хорьков с мал...   
759867  Vitapol Smakers Box - это лакомство с фруктами...   
759868  Versele-Laga Complete Ferret ВЕРСЕЛЕ-ЛАГА КОМП...   

                main_category  \
0       Компьютеры и ноутбуки   
1             Товары для дома   
2             Товары для дома   
3          Офис, школа, книги   
4             Товары для дома   
...                       ...   
759864        Товары для дома   
759865        Товары для дома   
759866        Товары для дома   
759867        Товары для дома   
759868        Товары для дома   

                                                    title  
0       Маршрутизатор LB-Link BL-WR450H WiFi Роутер дл...  
1       Кресло для геймеров Razer Iskur Green XL (RZ38...  
2       Кресло геймерское RZTK Cyber Red/Black (CBR01 RB)  
3       Набор Buromax Super Start для работы с маркерн...  
4       Кресло игровое Barsky Sport Drive Game Red (SD...  
...                                                   ...  
759864  Лакомство Gimbi Микс для грызунов 50 г травы и...  
759865  Лакомство Gimbi Микс для грызунов 50 г крапива...  
759866  Лакомство для хорьков Beaphar Ferret Bits 35 г...  
759867  Набор лакомств для грызунов Vitapol Smakers Bo...  
759868  Корм для хорьков Versele-Laga Complete Ferret ...  

[759869 rows x 5 columns]

### Check missing values

In [18]:
nan_values = df.isna()
nan_columns = nan_values.any()
nan_columns

deep_category              False
deep_category_minus_one    False
description                 True
main_category              False
title                      False
dtype: bool

In [19]:
df.description.isna().sum()

48402

In [20]:
df[df.description.isna()]

deep_category deep_category_minus_one description  \
252           Геймерские кресла Hator       Геймерские кресла        None   
320           Геймерские кресла Hator       Геймерские кресла        None   
375     Геймерские кресла Special4You       Геймерские кресла        None   
379             Геймерские кресла MSI       Геймерские кресла        None   
381           Геймерские кресла Hator       Геймерские кресла        None   
...                               ...                     ...         ...   
759784       Корма для рептилий Tetra      Корма для рептилий        None   
759787       Корма для рептилий Tetra      Корма для рептилий        None   
759796       Корма для рептилий Tetra      Корма для рептилий        None   
759798       Корма для рептилий Tetra      Корма для рептилий        None   
759812       Корма для рептилий Tetra      Корма для рептилий        None   

          main_category                                              title  
252     Товары для дома  Кресло игровое Hator Sport Essential Stealth (...  
320     Товары для дома  Кресло игровое Hator Arc S Mineral Grey (HTC-1...  
375     Товары для дома  Кресло игровое Special4You ExtremeRace black/p...  
379     Товары для дома                Кресло для геймеров MSI MAG CH120 I  
381     Товары для дома  Кресло игровое Hator Arc S Marrakesh Brown (HT...  
...                 ...                                                ...  
759784  Товары для дома  Корм для водоплавающих черепах Tetra ReptoDeli...  
759787  Товары для дома  Корм для водоплавающих черепах Tetra Gammarus ...  
759796  Товары для дома        Корм для земноводных Tetra ReptoFrog 100 мл  
759798  Товары для дома  Корм для водоплавающих черепах Tetra Gammarus ...  
759812  Товары для дома  Корм для водоплавающих черепах Tetra ReptoDeli...  

[48402 rows x 5 columns]

In [21]:
df[df.description.isna()].deep_category.value_counts()

Аксессуары для сублимации My Print               439
Аксессуары по уходу за автомобилем               374
Игровые консоли и детские приставки Microsoft    306
Трековые и струнные системы Skarlat              294
Ванны Besco                                      288
                                                ... 
Многофункциональные инструменты Makita             1
Циркулярные пилы (Дисковые)                        1
Дрели и миксеры Зенит                              1
Шуруповерты Grand                                  1
Корма для рептилий Cristel                         1
Name: deep_category, Length: 8332, dtype: int64

In [22]:
df[df.description.isna()].main_category.value_counts()

Сантехника и ремонт                       14611
Инструменты и автотовары                   7161
Компьютеры и ноутбуки                      4527
Товары для дома                            3452
Смартфоны, ТВ и электроника                3277
Бытовая техника, интерьер                  2794
Товары для детей                           2444
Алкогольные напитки и продукты питания     2393
Товары для бизнеса                         2064
Спорт и увлечения                          1569
Красота и здоровье                         1412
Офис, школа, книги                         1261
Дача, сад, огород                          1079
Одежда, обувь и аксессуары                  212
Подарки и товары для праздников             142
Автотовары                                    4
Name: main_category, dtype: int64

Загалом, кількість пустих дескрипшенів невелика, особливо якщо розглядати в контексті найглибшої категорії. Замість видалення, додамо тайтл в дескрипшн

In [27]:
df['description'] = df['description'].fillna(df['title'])

In [28]:
df[df.description.isna()]

Empty DataFrame
Columns: [deep_category, deep_category_minus_one, description, main_category, title]
Index: []

### Data analysys

In [2]:
df.describe()
%matplotlib inline

In [3]:
df.main_category.value_counts()

Сантехника и ремонт                       167337
Инструменты и автотовары                   78864
Спорт и увлечения                          64044
Товары для дома                            61613
Товары для детей                           58699
Красота и здоровье                         53984
Компьютеры и ноутбуки                      47505
Смартфоны, ТВ и электроника                44924
Алкогольные напитки и продукты питания     43725
Товары для бизнеса                         38161
Бытовая техника, интерьер                  34085
Офис, школа, книги                         28841
Дача, сад, огород                          21203
Одежда, обувь и аксессуары                  7996
Подарки и товары для праздников             4000
Сексшоп                                     3998
Услуги и сервисы                             886
Автотовары                                     4
Name: main_category, dtype: int64

In [4]:
df.deep_category.value_counts()

Люстры Brille                                   2649
Бра и настенно-потолочные светильники Brille    2397
Полочные стеллажи для склада Smartzone          2188
Надувная мебель и аксессуары Intex              1805
Детские электро- и веломобили Bambi             1771
                                                ... 
Гипсокартон                                        1
Доски для рисования Kenvo                          1
Тихое вино Primavoce                               1
Тихое вино Chateau Meillier                        1
Корма для грызунов Dajana                          1
Name: deep_category, Length: 80047, dtype: int64

In [5]:
df.deep_category_minus_one.value_counts()

Компьютеры, неттопы, моноблоки           5508
Мониторы                                 4601
Люстры                                   4293
Бра и настенно-потолочные светильники    4190
Ноутбуки                                 4110
                                         ... 
Наливные полы                               1
Лента звукоизоляционная                     1
Крепкие напитки                             1
Подарочные визитницы                        1
Копировальные аппараты                      1
Name: deep_category_minus_one, Length: 2840, dtype: int64

Ми маємо 18 основних категорій, 2840 перед найглибших та 80047 найглибших

In [6]:
df['deep_cat_in_deep_minus_one'] = df.apply(lambda x: x.deep_category_minus_one in x.deep_category, axis=1)

In [8]:
df['deep_cat_in_deep_minus_one'].value_counts()

True     682927
False     76942
Name: deep_cat_in_deep_minus_one, dtype: int64

In [12]:
df.loc[df['deep_cat_in_deep_minus_one'] == True][['deep_category' , 'deep_category_minus_one']]

deep_category         deep_category_minus_one
0                       Маршрутизаторы LB-LINK                  Маршрутизаторы
1                      Геймерские кресла Razer               Геймерские кресла
2                       Геймерские кресла RZTK               Геймерские кресла
3       Магниты для досок и аксессуары Buromax  Магниты для досок и аксессуары
4                     Геймерские кресла Barsky               Геймерские кресла
...                                        ...                             ...
759864                Корма для грызунов Gimbi              Корма для грызунов
759865                Корма для грызунов Gimbi              Корма для грызунов
759866              Корма для грызунов Beaphar              Корма для грызунов
759867              Корма для грызунов Vitapol              Корма для грызунов
759868         Корма для грызунов Versele-Laga              Корма для грызунов

[682927 rows x 2 columns]

Ми бачимо, що в більшості випадків остання і перед остання категорія відрізняється лише наявністью бренду виробника в назві.